janusgraph-core/src/main/java/org/janusgraph/diskstorage/log/kcvs/KCVSLog.java

```diff
@@ -777,6 +791,10 @@ public void run() {
                 if (e.getCause() instanceof PermanentBackendException) {
                     throw e;
                 }
+                if(Thread.currentThread().isInterrupted()){
+                    log.warn("Could not read messages for timestamp ["+messageTimeStart+"]. Thread was interrupted. Thus, no mo retries will be conducted.",e);
+                    throw e;
```

Comment:
Is it possible to test this?

Reply:
I will wait for CI to pass in #2757 and merge it. After that, there will be available a `TestLogger` in tests which we can use to intercept logs. I think we could than try to start backend, shutdown storage and close backend after that. I think in that case we will be able to see this log as KCVSLog will not be closed properly after storage shutdown (due to inability to communicate with the storage backend). I will try to reproduce it with Cassandra

Reply:
After working on #2769 I couldn't reproduce this issue anymore. Most likely this was due to underlying storages improper shutdown.